mautic/mautic
# Deliver contact created webhook event when a contact is identified rather than visitor

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

The contact created webhook was falsely advertised. Short of contacts created manually or imported, almost all contacts start as visitors. That's when the webhook is delivered. A contact is someone identified, not a visitor. A contact in Mautic is someone who has been identified versus a visitor/anonymous contact. This change will only deliver an event if a contact just became identified meaning it was converted from a visitor to a contact. 

This is a behavior change but matches Mautic's own definition of a true contact. 

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create a new contact event webhook to a requestb.in
2. Create a landing page
3. View the landing page in while not logged into Mautic after clearing the browser's cookies (so a new visitor is created)
4. Note the webhook delivers a useless event for the visitor

#### Steps to test this PR:
1. Embed a form that maps email address on the landing page
2. Clear browsers cookies
3. Visit the landing page while not logged into Mautic
4. Submit the form with an email address
5. Review the request bin and note that the conversion from visitor to contact (i.e. a new contact) is represented where the first landing page visit is not
